blog/_posts/2020-04-20-first-release.adoc

```diff
@@ -0,0 +1,211 @@
+This is useful in combination with:
+
+Parent module (no default keybinding)::
+This action brings to the `mod` declaration which declared the current module.
+It doesn't have a shortcut assigned by default, as there's no corresponding built-in action, but it is highly recommended to assign one.
+
+Workspace Symbol kbd:[Ctrl+T]::
+This is probably the shortcut I use most often.
+It is a fuzzy-search interface for all "symbols" (structs, enums, functions, field) in the project, its dependencies and the standard library.
+The search tries to be smart, in that, by default, it looks only for types in your project, and, failing that, for functions.
+It is possible to force search in dependencies by adding `#` to the query string, and search for all symbols by adding `*`.
+Unfortunately, this doesn't work in VS Code at the moment, as it stopped passing these symbols to the language server since the last update.
+
+video::/assets/blog/first-release/workspace-symbol.webm[options="autoplay,loop",width=100%]
+
+Document Symbol kbd:[Ctrl+Shift+O]::
+Like workspace symbol, but for things in the current file.
+The same underlying LSP request powers file outline and breadcrumbs.
+
+image::https://user-images.githubusercontent.com/1711539/80090645-1e4def00-8560-11ea-901d-d1cdc0ab8f50.png[]
+
+Implementation kbd:[Ctrl+F12]::
+This shortcut works on structs, enums and traits, and will show you the list of corresponding impls.
+
+Syntax Highlighting::
+While not exactly about navigation, semantic syntax highlighting helps with reading code.
+Rust analyzer underlines mutable variables, distinguishes between modules, traits and types and
+provides helpful type and parameter hints.
+
+image::https://user-images.githubusercontent.com/1711539/80091615-b5677680-8561-11ea-82de-e1517e4fef18.png[]
+
+Run (no default keybinding)::
+After navigation, the feature I use most is probably the **Run** button.
+This action runs the test function, test module or main function at the given cursor position.
+It is also available as a code-lens, but I personally exclusively use kbd:[ctrl+r] for it, as I need this action all the time.
+What's more, with a short cut you can re-run the last command, which is hugely useful when you are debugging a failing test.
+This action is pretty smart in that it does the following things for you:
+
+* determines the appropriate `--package` argument for `Cargo`,
+* uses the full path to the test, including the module,
+* sets the `--no-capture` argument, so that debug prints are visible,
+* sets the `RUST_BACKTRACE` environmental variable, so that you don't have to re-run on panic.
+
+Sadly, such context-dependent run configurations are not a part of the LSP protocol yet, so this feature is implemented using a custom protocol extension.
+
+video::/assets/blog/first-release/run.webm[options="autoplay,loop",width=100%]
+
+Punctuation-aware code completion::
+I sometimes write code as well, and rust-analyzer has a couple of feature in this department.
```

Comment:
```suggestion
I sometimes write code as well, and rust-analyzer has a couple of features in this department.
```

Reply:
gosh, I've botched a rebase, should be better now